Los ficheros binarios vzctl-3.0.24.2/a.out y vzctl-3.0.24.2-iproute2/a.out son distintos
diff -uNr vzctl-3.0.24.2/src/lib/config.c vzctl-3.0.24.2-iproute2/src/lib/config.c
--- vzctl-3.0.24.2/src/lib/config.c	2010-09-06 16:25:28.000000000 +0200
+++ vzctl-3.0.24.2-iproute2/src/lib/config.c	2010-10-10 17:22:30.000000000 +0200
@@ -820,7 +820,7 @@
 		if (inet_ntop(family, ip, dst, sizeof(dst)) == NULL)
 			return ERR_INVAL;
 		if (!find_ip(&net->ip, dst))
-			add_str_param(&net->ip, dst);
+			add_str_param(&net->ip, val);
 	}
 
 	return 0;
diff -uNr vzctl-3.0.24.2/src/lib/util.c vzctl-3.0.24.2-iproute2/src/lib/util.c
--- vzctl-3.0.24.2/src/lib/util.c	2010-09-06 16:25:28.000000000 +0200
+++ vzctl-3.0.24.2-iproute2/src/lib/util.c	2010-10-10 17:10:02.000000000 +0200
@@ -289,12 +289,27 @@
 
 int get_netaddr(const char *ip_str, unsigned int *ip)
 {
-	if (strchr(ip_str, ':')) {
-		if (inet_pton(AF_INET6, ip_str, ip) <= 0)
+	char *ip_address = NULL;
+	int slash_pos = 0;
+	
+	if (0 != (slash_pos = strcspn(ip_str, "/"))) {
+		ip_address = malloc(slash_pos + 1);
+		strncpy(ip_address, ip_str, slash_pos);
+		ip_address[slash_pos + 1] = '\0';
+	} else {
+		size_t ip_address_len = strlen(ip_str);
+		ip_address = malloc(ip_address_len + 1);
+		strncpy(ip_address, ip_str, ip_address_len);
+		ip_address[ip_address_len + 1] = '\0';
+	}
+	
+	if (strchr(ip_address, ':')) {
+		if (inet_pton(AF_INET6, ip_address, ip) <= 0)
 			return -1;
 		return AF_INET6;
 	}
-	if (inet_pton(AF_INET, ip_str, ip) <= 0)
+	
+	if (inet_pton(AF_INET, ip_address, ip) <= 0)
 		return -1;
 	return AF_INET;
 }
